projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -194,11 +195,39 @@ public function capture_product_view() {
 		);
 	}
 
+	/**
+	 * Get the selected shipping option for a cart item. If the name cannot be found in the options table, the method's
+	 * ID will be used.
+	 *
+	 * @param string $cart_item_key the cart item key.
+	 *
+	 * @return mixed|void
+	 */
+	public function get_shipping_option_for_item( $cart_item_key ) {
+		$packages         = wc()->shipping()->get_packages();
+		$selected_options = wc()->session->get( 'chosen_shipping_methods' );
+		foreach ( $packages as $package_id => $package ) {
```

Comment:
I'm not familiar with those methods or how they can be extended by other Woo extensions, but I was wondering if we should be extra defensive before to do anything with the outcome of `$packages` and `$selected_options`?

Reply:
I don't think `get_packages` can be extended, and it just returns an empty array if nothing found, but in https://github.com/Automattic/jetpack/pull/33544/files/d8ac52ee6c7bac6d01891860dcd9f353a0bd1435..f7e34f92b7c6cf9858d7f35bd520262e4dfa89ec I added some checks anyway - I think getting the session variable could be filtered so good idea on that one. Thanks!